_posts/2020-18-06-Triplestores-Onotologies-Reasoning.md

```diff
@@ -0,0 +1,354 @@
+specification if the Triplestore supports it.  Additionally, if two
+predicates define different classes for their respective URLs, a user
+can end up in a position where a reasoner infers a `Person` class and
+a `Cat` class on the same URL, for instance, which will confuse a user
+who expecting a `Person` and a `Cat` to be distinct individuals.  The
+curators need to be careful when using predicates to ensure they do
+not cause odd or confusing inferences to be made.
+
+However, there are even more powerful features of OWL 2.  Predicates
+that can be defined as
+[_transitive_](https://www.w3.org/TR/2012/REC-owl2-primer-20121211/#Property_Characteristics)
+in OWL which means that if a predicate connects A and B while B uses
+the same predicate to connect B to C then A has the same effect as if
+A also connected to C.
+
+An instance from IrishGen should make this more clear.
+
+```turtle
+<#Laidcnén>
+    a foaf:Person;
+    irishRel:nomName "Laidcnén";
+    rel:childOf <#Cummine>;
+    irishRel:ancestorOfGroup <#h-ÚiBuide>.
+```
+
+IrishGen heavily uses the Relationship ontology which defines
+`rel:parentOf` and `rel:childOf`.  These are defined as a
+`rdf:subPropertyOf` `rel:ancestorOf` and `rel:descendantOf`.
+`rel:ancestor` and `rel:descendantOf` are themselves defined as
+transitive and the inverse of each other.  Summing all of that up, it
+means that a reasoner can completely reconstruct a genealogical line
+without human intervention.  An instance of this will be illustrative.
+Without needing to encode by hand Laidcnén has 2301 ancestors.
+
+This is incredibly powerful.  It means that only a minimum of
+information needs to be encoded or curated by a human and the benefit
+of inferring new information about existing information in the
+Triplestore.  This is completely unlike a relational database which
+needs all the information encoded or inferred by logic by computer
+systems written by humans.
+
+This singular feature of Linked Data was the driver behind the
+decision to use Triplestores and RDF rather than conventional
+relational or other graph databases.  Massive amounts of human power
+would be necessary to encode all the information not only directly
+stated but also logically implied by the genealogies.  That level of
+data extraction and encoding by team of humans who are experts in the
+medieval Irish genealogical tradition, even a large team assisted by
+specialised technology and automation, would be prohibitively
+expensive.  In essence, any assisting automation would by necessity
+reinvent Linked Data rather than just taking advantage of an already
+existing technology.
+
+## Triplestores and RDF strategies
+
+Now that the reader is acquainted with inferrencing, it would be good
+to discuss inferrencing strategies.  While this may seem like a very
+esoteric topic, it is critical to the experience of writing and
+running queries on Triplestores and will effect the perception of
+usefulness of Triplestores in general.
```

Comment:
Perhaps make it clear at this point that this isn't about the user coming up with their own inferencing strategy but about how different triplestores work.